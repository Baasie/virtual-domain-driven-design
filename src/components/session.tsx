import React, { FC } from "react"
import tw from "twin.macro"

export interface SessionContent {
  id: string
  date: string
  level: string
  tags: string[]
  title: string
  video: string
}

interface SessionProps {
  session: SessionContent
}

const Session: FC<SessionProps> = ({ session }) => {
  return (
    <div tw="bg-white w-96 rounded-lg shadow-md p-2 m-2 flex flex-col">
      <div tw="text-sm text-gray-600">{session.date}</div>
      <div css={[{ paddingTop: "56.25%" }, tw`relative`]}>
        <iframe
          title={session.title}
          tw="absolute top-0 left-0 w-full h-full"
          allowFullScreen={true}
          src={session.video}
          scrolling="no"
          frameBorder={0}
        ></iframe>
      </div>
      <a
        tw="text-sm text-left font-bold text-blue-600 hover:cursor-pointer hover:text-blue-400"
        href={session.video}
        target="_blank"
        rel="noopener noreferrer"
      >
        {session.title}
      </a>
      <div tw="px-1 w-full flex flex-row flex-wrap">
        <div tw="flex-shrink-0 leading-none text-xs tracking-tighter bg-blue-700 text-white rounded-md p-1 m-1">
          Level: {session.level}
        </div>
        {session.tags.map((tag, index) => {
          return (
            <div
              key={index}
              tw="flex-shrink-0 leading-none text-xs tracking-tighter bg-gray-200 text-gray-700 rounded-md p-1 m-1"
            >
              {tag}
            </div>
          )
        })}
      </div>
    </div>
  )
}

export default Session
