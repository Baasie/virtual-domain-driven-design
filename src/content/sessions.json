{
  "upcoming_sessions": [
    {
      "title": "[Panel] What can we learn from open-source with Matteo Collina",
      "date": "Tuesday, August 4th. 2020",
      "time": "18:30 CEST",
      "img": "https://secure.meetupstatic.com/photos/event/5/7/8/8/highres_491542408.jpeg",
      "video": null,
      "embedded": null,
      "podcast": null,
      "description": "Thanks to Krisztina we will have Matteo Collina as a special guest on our next panel. Matteo is a long time Nodejs contributor and TSC member. Open-source software is a success story, and undoubtedly one, we can learn from. In OSS the clocks tick differently, but it is software built for users, to solve problems - both relatively unknowns factors at the beginning. So what can DDD developers for businesses learn from that experience: how to handle these uncertainties,  how is the Ubiquitous Language developed in the Open source world? How do you do design in OSS? And many more questions!",
      "links": [
        {
          "label": "Panel on Zoom",
          "url": "https://us02web.zoom.us/j/89077595529"
        },
        {
          "label": "YouTube live stream",
          "url": "https://youtu.be/F4SwUssweGY"
        },
        {
          "label": "The Event on Meetup",
          "url": "https://www.meetup.com/Virtual-Domain-Driven-Design-meetup/events/272201468/"
        }
      ]
    },
    {
      "title": "[Test] What can we learn from open-source with Matteo Collina",
      "date": "Tuesday, August 4th. 2020",
      "time": "18:30 CEST",
      "img": "https://secure.meetupstatic.com/photos/event/5/7/8/8/highres_491542408.jpeg",
      "video": null,
      "embedded": null,
      "podcast": null,
      "description": "Thanks to Krisztina we will have Matteo Collina as a special guest on our next panel. Matteo is a long time Nodejs contributor and TSC member. Open-source software is a success story, and undoubtedly one, we can learn from. In OSS the clocks tick differently, but it is software built for users, to solve problems - both relatively unknowns factors at the beginning. So what can DDD developers for businesses learn from that experience: how to handle these uncertainties,  how is the Ubiquitous Language developed in the Open source world? How do you do design in OSS? And many more questions!",
      "links": [
        {
          "label": "Panel on Zoom",
          "url": "https://us02web.zoom.us/j/89077595529"
        },
        {
          "label": "YouTube live stream",
          "url": "https://youtu.be/F4SwUssweGY"
        },
        {
          "label": "The Event on Meetup",
          "url": "https://www.meetup.com/Virtual-Domain-Driven-Design-meetup/events/272201468/"
        }
      ]
    }
  ]
}
