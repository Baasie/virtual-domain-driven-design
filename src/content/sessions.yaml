---
sessions:
  - title: "[Panel] One user to bind them all? Products, teams and bounded contexts"
    date: "Wednesday, July 29th. 2020"
    time: "19:00 CEST"
    video: "https://www.youtube.com/embed/wMX9PdRYuRo"
    podcast: "https://oembed.libsyn.com/embed?item_id:15410591"
    description:
      "In the last meetup, Krisztina found something Jessica said interesting to dive into:
      'We are talking in DDD about Bounded Contexts and independent teams and applications,
      but then we all coupled by having the same user.' That statement led to an exciting
      dialogue by inviting Dawn: 'It is not so much coupling at the user as that sounds as
      if the user has to fit into what has built but starting with the user in the centre or
      intersection of the various 'contexts'. If your context no longer aligns with the user,
      who are you problem-solving/building the solution for?'
      Joining us with Krisztina, Jessica and Dawn will be Manuel from the Book team topologies,
      and together we will explore heuristics to organise teams, and their interaction and design
      bounded context for fast flow, which each serve the same customer."
    links:
      - label: "The Event on Meetup"
        url: "https://www.meetup.com/Virtual-Domain-Driven-Design-meetup/events/272058374/"
  - title: "[Hands-on] Speaking truth to power: practice argumentation with Diana"
    date: "Tuesday, July 21th. 2020"
    time: "19:00 CEST"
    video:
    podcast:
    description:
      "If you could influence one impactful decision today, what would it be? If you could be heard
      and trusted for 10 minutes, what would you say? To whom? More importantly,
      how do you transform recommendations into well-reasoned decisions arrived at collectively?
      In this workshop, we’ll strengthen our ability to craft a strategic argument that gets heard.
      We’ll work together to strengthen a position and the reasons that support it.
      We’ll explore the fundamental skills required and brainstorm follow-up activities."
    links:
      - label: "The Event on Meetup"
        url: "https://www.meetup.com/Virtual-Domain-Driven-Design-meetup/events/271938834/"
