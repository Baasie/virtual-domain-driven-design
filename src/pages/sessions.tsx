import React, { ReactElement } from "react"
import { graphql, useStaticQuery } from "gatsby"
import UpcomingSession from "../components/upcoming-session"

import NoUpcomingImg from "../images/no_upcoming.svg"
import Layout from "../templates/layout"

import Session from "../components/session"

const UpcomingSessions = ({ sessions }) => {
  if (sessions.length > 0) {
    return (
      <div className="my-8 w-4/5 lg:w-2/3 xl:w-1/2">
        <h2>Upcoming Sessions</h2>
        <div className="flex flex-col items-center justify-start">
          {sessions.map((session, index) => {
            return (
              <UpcomingSession key={index} session={session}></UpcomingSession>
            )
          })}
        </div>
      </div>
    )
  }
  return (
    <div className="my-8 w-4/5 lg:w-2/3 xl:w-1/2">
      <h2>Upcoming Sessions</h2>
      <div className="flex flex-col items-center justify-start">
        <NoUpcomingImg className="h-64 mb-6" />
        <div>More sessions are coming to you eventually consistent...</div>
        <a
          className="p-4 mt-6 bg-blue-400 floating-action-button text-white"
          href="https://sessionize.com/virtual-ddd-meetup"
          target="_blank"
          rel="noopener noreferrer"
        >
          Propose a session
        </a>
      </div>
    </div>
  )
}

const PastSessions = ({ sessions }) => {
  if (sessions.length > 0) {
    return (
      <div className="w-full flex flex-col items-center justify-start">
        <h2 className="my-6 w-4/5 lg:w-2/3 xl:w-1/2">Past Sessions</h2>
        <div className="w-11/12 md:w-5/6">
          <div className="flex items-stretch justify-center flex-wrap">
            {sessions.map((session, index) => {
              return <Session key={index} session={session} />
            })}
          </div>
        </div>
      </div>
    )
  }
  return <div></div>
}

function Sessions(): ReactElement {
  const data = useStaticQuery(graphql`
    query {
      upcoming: allContentYaml(
        filter: { upcomingSessions: { elemMatch: { title: { ne: null } } } }
      ) {
        nodes {
          upcomingSessions {
            date
            description
            img
            time
            title
            links {
              label
              url
            }
          }
        }
      }
      past: allContentYaml(
        filter: { sessions: { elemMatch: { title: { ne: null } } } }
      ) {
        nodes {
          sessions {
            title
            date
            level
            video
            tags
          }
        }
      }
    }
  `)
  return (
    <Layout>
      <div className="section" id="Sessions">
        <UpcomingSessions
          sessions={data.upcoming.nodes[0].upcomingSessions}
        ></UpcomingSessions>
        <PastSessions
          sessions={data.past.nodes[0].sessions.filter(
            (session) => session.video
          )}
        ></PastSessions>
      </div>
    </Layout>
  )
}

export default Sessions
