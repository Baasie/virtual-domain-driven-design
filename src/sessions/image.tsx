import { graphql } from "gatsby"

export const upcomingImage = graphql`
  fragment upcomingImage on ContentYamlUpcomingSessions {
    img {
      publicURL
      childImageSharp {
        fluid {
          ...GatsbyImageSharpFluid
        }
      }
    }
  }
`

export const sessionImage = graphql`
  fragment sessionImage on ContentYamlSessions {
    img {
      publicURL
      childImageSharp {
        fluid {
          ...GatsbyImageSharpFluid
        }
      }
    }
  }
`
