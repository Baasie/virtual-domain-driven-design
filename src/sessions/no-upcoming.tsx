// @ts-ignore
import NoUpcomingImg from "../images/no_upcoming.svg"
import ThreeDBlueButton from "../components/three-d-blue-button"
import React from "react"

const NoUpcoming = () => (
  <div tw="my-8 w-4/5 lg:w-2/3 xl:w-1/2">
    <h2 tw="my-2 w-4/5 lg:w-2/3 xl:w-1/2 text-blue-800 text-3xl">
      Upcoming Sessions
    </h2>
    <div tw="flex flex-col items-center justify-start">
      <NoUpcomingImg tw="h-64 mb-6" />
      <div tw="my-4">
        No upcoming session finalized yet
      </div>
      <ThreeDBlueButton href="https://feedback.userreport.com/a15e4e61-2323-40a1-90b4-1267e010e35c/">
        Propose a session
      </ThreeDBlueButton>
    </div>
  </div>
)

export default NoUpcoming
