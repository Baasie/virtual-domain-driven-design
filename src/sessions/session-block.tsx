import React from "react"
import tw from "twin.macro"

import SessionLink from "./../sessions/upcoming-session"
import ThreeDBlueButton from "../components/core/three-d-blue-button"

type SessionBlockProps = {
  description: string
  title: string
  links?: typeof SessionLink[]
  video: string
}

const SessionBlock = (sessionBlockProps: SessionBlockProps) => {
  return (
    <div tw="flex xl:flex-row flex-col items-center lg:w-2/3 w-5/6 m-4">
      <div tw="xl:w-2/3 w-full m-4 ">
        <div tw="text-center xl:w-4/5 rounded-lg shadow-md p-2">
          {sessionBlockProps.description}
        </div>
      </div>
      <div tw="xl:w-1/3 w-full rounded-lg shadow-md p-1 m-2">
        <div css={[{ paddingTop: "56.25%" }, tw`relative`]}>
          <iframe
            title={sessionBlockProps.title}
            tw="absolute top-0 left-0 w-full h-full"
            allowFullScreen={true}
            src={sessionBlockProps.video}
            scrolling="no"
            frameBorder={0}
          ></iframe>
        </div>
        {sessionBlockProps.links && (
          <div tw="m-4 sm:space-x-4 space-y-2 sm:space-y-0 flex flex-col sm:flex-row items-center justify-center">
            {sessionBlockProps.links.map((link) => {
              return (
                <ThreeDBlueButton tw="text-xs " href={link.url}>
                  {link.label}
                </ThreeDBlueButton>
              )
            })}
          </div>
        )}
      </div>
    </div>
  )
}

export default SessionBlock
