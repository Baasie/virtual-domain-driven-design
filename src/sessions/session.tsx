import { graphql } from "gatsby"
import Img from "gatsby-image"
import React, { FC } from "react"
import "twin.macro"
import ThreeDBlueButton from "../components/three-d-blue-button"

export const session = graphql`
  fragment session on ContentYaml {
    sessions {
      id
      title
      date
      time
      description
      level
      tags
      video
      podcast
      ...sessionImage
    }
  }
`

export enum ContentLevel {
  All = "all",
  Beginner = "beginner",
  Intermediate = "intermediate",
  Advanced = "advanced",
}

export type SessionContent = {
  id: string
  date: string
  title: string
  time: string
  description: string
  level: ContentLevel
  tags: string[]
  video: string
  podcast?: string
  img: any
}

type SessionProps = {
  session: SessionContent
}

const Session: FC<SessionProps> = ({ session }) => {
  const linkToSession = "/sessions/" + session.id
  return (
    <div tw="md:flex shadow-lg mx-6 md:mx-auto my-4 max-w-lg lg:max-w-2xl h-96 md:h-64">
      <Img
        tw="h-full w-full md:w-1/3 object-cover rounded-lg rounded-r-none h-2/5 md:h-64"
        fluid={session.img.childImageSharp.fluid}
        alt="bag"
      />
      <div tw="w-full md:w-2/3 px-4 py-2 bg-white rounded-lg h-3/5 md:h-64">
        <h2 tw="text-base md:text-xl text-gray-800 font-medium mr-auto">
          {session.title}
        </h2>
        <p tw="text-xs md:text-sm text-gray-700 mt-4 break-all line-clamp-3 lg:line-clamp-4">
          {session.description}
        </p>
        <div tw="flex items-center mt-2 mb-2 h-14 overflow-hidden">
          <div tw="flex flex-wrap w-3/5 h-14">
            <div tw="flex-shrink-0 leading-none text-xs bg-blue-700 text-white rounded-md p-1 m-1 h-6">
              Level: {session.level}
            </div>
            {session.tags.map((tag, index) => {
              return (
                <div
                  key={index}
                  tw="flex-shrink-0 leading-none text-xs bg-gray-200 text-gray-700 rounded-md p-1 m-1 h-6"
                >
                  {tag}
                </div>
              )
            })}
          </div>
          <div tw="w-2/5 flex">
            <ThreeDBlueButton tw="flex-shrink-0 " to={linkToSession}>
              More info
            </ThreeDBlueButton>
          </div>
        </div>
      </div>
    </div>
  )
}

export default Session
