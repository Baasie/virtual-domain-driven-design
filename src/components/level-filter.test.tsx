import React from "react"
import renderer from "react-test-renderer"
import "jest-styled-components"
import { fireEvent, render } from "@testing-library/react"

// This below import is what gives us the "toBeInTheDocument" method
import "@testing-library/jest-dom/extend-expect"
import LevelFilter from "./level-filter"
import { ContentLevel } from "../sessions/session"

describe("The LevelFilter", () => {
  beforeEach(() => jest.resetAllMocks())

  const setLevelFilter = jest.fn()
  it("should create or check the snapshot", () => {
    const container = renderer
      .create(<LevelFilter setLevelFilter={setLevelFilter} />)
      .toJSON()

    expect(container).toMatchSnapshot()
  })

  it("should render one button for each level", () => {
    const { queryAllByRole } = render(
      <LevelFilter setLevelFilter={setLevelFilter} />
    )
    const result = queryAllByRole("button")

    //TODO test if the right button is active
    expect(result).toHaveLength(4)
  })

  it("should set the correct level with each click", () => {
    const { queryAllByRole } = render(
      <LevelFilter setLevelFilter={setLevelFilter} />
    )
    const buttons = queryAllByRole("button")
    const all = buttons[0]
    const beginner = buttons[1]

    fireEvent.click(all)
    expect(setLevelFilter).toHaveBeenCalledWith([
      ...Object.values(ContentLevel),
    ])

    fireEvent.click(beginner)
    expect(setLevelFilter).toHaveBeenCalledWith([
      ContentLevel.All,
      ContentLevel.Beginner,
    ])
  })
})
