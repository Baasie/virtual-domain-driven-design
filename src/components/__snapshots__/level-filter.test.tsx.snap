// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The LevelFilter should create or check the snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  font-size: 1.25rem;
  line-height: 1.75rem;
  margin: 0.5rem;
  padding: 0.5rem;
  --tw-text-opacity: 1;
  color: rgba(255,255,255,var(--tw-text-opacity));
  border-radius: 0.5rem;
  border-width: 2px;
  --tw-border-opacity: 1;
  border-color: rgba(59,130,246,var(--tw-border-opacity));
  --tw-bg-opacity: 1;
  background-color: rgba(29,78,216,var(--tw-bg-opacity));
}

.c1:hover {
  --tw-border-opacity: 1;
  border-color: rgba(96,165,250,var(--tw-border-opacity));
}

.c2 {
  font-size: 1.25rem;
  line-height: 1.75rem;
  margin: 0.5rem;
  padding: 0.5rem;
  --tw-text-opacity: 1;
  color: rgba(255,255,255,var(--tw-text-opacity));
  border-radius: 0.5rem;
  border-width: 2px;
  --tw-border-opacity: 1;
  border-color: rgba(59,130,246,var(--tw-border-opacity));
}

.c2:hover {
  --tw-border-opacity: 1;
  border-color: rgba(96,165,250,var(--tw-border-opacity));
}

.c3 {
  font-size: 1.25rem;
  line-height: 1.75rem;
  margin: 0.5rem;
  padding: 0.5rem;
  --tw-text-opacity: 1;
  color: rgba(255,255,255,var(--tw-text-opacity));
  border-radius: 0.5rem;
  border-width: 2px;
  --tw-border-opacity: 1;
  border-color: rgba(59,130,246,var(--tw-border-opacity));
}

.c3:hover {
  --tw-border-opacity: 1;
  border-color: rgba(96,165,250,var(--tw-border-opacity));
}

.c4 {
  font-size: 1.25rem;
  line-height: 1.75rem;
  margin: 0.5rem;
  padding: 0.5rem;
  --tw-text-opacity: 1;
  color: rgba(255,255,255,var(--tw-text-opacity));
  border-radius: 0.5rem;
  border-width: 2px;
  --tw-border-opacity: 1;
  border-color: rgba(59,130,246,var(--tw-border-opacity));
}

.c4:hover {
  --tw-border-opacity: 1;
  border-color: rgba(96,165,250,var(--tw-border-opacity));
}

<div
  className="c0"
>
  <button
    className="c1"
    onClick={[Function]}
  >
    All levels
  </button>
  <button
    className="c2"
    onClick={[Function]}
  >
    Beginner
  </button>
  <button
    className="c3"
    onClick={[Function]}
  >
    Intermediate
  </button>
  <button
    className="c4"
    onClick={[Function]}
  >
    Advanced
  </button>
</div>
`;
